File: python-backend/ModuAgent/components/reasoning/llm/qwen.py
from __future__ import annotations

import os
from typing import Optional

from components.reasoning.llm.base_llm import BaseLLMReasoner

_DEFAULT_BASE_URL = "https://dashscope.aliyuncs.com/compatible-mode/v1"
_DEFAULT_MODEL = "qwen-max"


class QwenLLMReasoner(BaseLLMReasoner):
    def __init__(
        self,
        api_key: Optional[str] = None,
        base_url: Optional[str] = None,
        default_model: Optional[str] = None,
        timeout: float = 120.0,
        system_prompt: Optional[str] = None,
    ) -> None:
        resolved_key = api_key or os.getenv("MODU_QWEN_API_KEY", "")
        resolved_url = base_url or os.getenv("MODU_QWEN_BASE_URL", _DEFAULT_BASE_URL)
        resolved_model = default_model or os.getenv("MODU_QWEN_MODEL", _DEFAULT_MODEL)
        super().__init__(
            api_key=resolved_key,
            base_url=resolved_url,
            default_model=resolved_model,
            timeout=timeout,
            system_prompt=system_prompt,
        )

File: packages/modu-agent/src/reasoning/llm/qwen.ts
// 对应 Python: components/reasoning/llm/qwen.py
// Qwen (通义千问) LLM 推理器
import { BaseLLMReasoner } from './base-llm.js'

const _DEFAULT_BASE_URL = 'https://dashscope.aliyuncs.com/compatible-mode/v1'
const _DEFAULT_MODEL = 'qwen-max'

/**
 * Qwen LLM 推理器。
 * 对应 Python QwenLLMReasoner。
 *
 * API key / base_url / model 解析优先级：
 *   显式参数 > MODU_QWEN_* 环境变量 > 默认值
 */
export class QwenLLMReasoner extends BaseLLMReasoner {
  constructor(
    apiKey?: string | null,
    baseUrl?: string | null,
    defaultModel?: string | null,
    timeout: number = 120.0,
    systemPrompt?: string | null,
  ) {
    const resolvedKey = apiKey || process.env.MODU_QWEN_API_KEY || ''
    const resolvedUrl = baseUrl || process.env.MODU_QWEN_BASE_URL || _DEFAULT_BASE_URL
    const resolvedModel = defaultModel || process.env.MODU_QWEN_MODEL || _DEFAULT_MODEL
    super(
      resolvedKey,
      resolvedUrl,
      resolvedModel,
      timeout,
      systemPrompt,
    )
  }
}
